src/App.js

```diff
@@ -23,19 +27,24 @@ export default class App extends Component {
           return item.id === data.id
             ? { ...item, quantity: item.quantity + 1 }
             : item
-        })
+        }),
+        modalStatus: true
       })
     } else {
       this.setState({
         cart: cart.concat(data),
-        productIds: productIds.concat(data.id)
+        productIds: productIds.concat(data.id),
+        modalStatus: true
       })
     }
   }
 
-  updateProductToCart = data => {
+  updateProductToCart = (data, idRemove) => {
+    const { productIds } = this.state
     this.setState({
-      cart: data
+      cart: data,
+      productIds: productIds.filter(item => item !== idRemove),
```

Comment:
let's rename this variable.
`productIds.filter(productId => productId != removedId`